tardis/simulation/base.py

```diff
@@ -410,3 +411,138 @@ def from_config(cls, config, **kwargs):
                    convergence_strategy=config.montecarlo.convergence_strategy,
                    nthreads=config.montecarlo.nthreads)
 
+    @classmethod
+    def from_hdf5(cls, file_path):
+        """
+        This function converts a hdf5 file to a Radial1DModel Object.
+
+        Parameters
+        ----------
+
+        file_path : `str`
+            path to Simulation generated hdf file
+
+        Returns
+        -------
+
+        model : `~Radial1DModel`
+        """
+
+        if file_path is None:
+            raise ValueError("File Path can`t be None")
+        if not os.path.exists(file_path):
+            raise IOError("Supplied HDF5 File %s does not exists" % file_path)
+
+        with h5py.File(file_path, 'r') as h5_file:
+            sim_dict = {}  # It will store HDF Objects in a dict form
+            for simulation in h5_file.keys():
+                for key in h5_file[simulation]:
+                    sim_dict[key] = {}
+                    if 'model' in key:
+                        cls.read_model_data(
+                            h5_file, simulation + '/model/', sim_dict[key], file_path)
```

Comment:
If you create a classmethod under `Radial1DModel` named `from_hdf` you could call this here like `Radial1DModel.from_hdf(hdf_file, ...)` and you'd get a fully populated Model. This way the `from_hdf` concept would be more modular and easier to extend.